### DISCUSSION TITLE
How to use enviroment variables outside <srcipt setup>

### DISCUSSION THREAD

**Discussion by gezichenshan** [nuxt/nuxt#16813] [Questions]
I read the doc that we can use `useRuntimeConfig` in <script setup> to import the environment variables from what I defined in .env and .env.staging file.

If I want to use env variables in staging when developing I can run `nuxt dev --mode staging`, which is good.

However I want to use them outside the vue component, for example a http-request.js file, could it be?

I found no methods listed in nuxt doc.

Then I found I can do it by using `import.meta.env` from [vite](https://vitejs.dev/guide/env-and-mode.html).

But running`nuxt dev --mode staging` will not make the nuxt project read the environment variables in .env.staging but .env.

Is there any solution?

**Comment by sevillaarvin** [comment]
What I currently do is I inject the environment variable from the page to my function:

service/api.js
```js
export const useApi = ({ baseURL }) => /* whatever */
```

pages/index.vue
```vue
<script setup>
import { useApi } from "@/service/api"

useApi({ baseURL: useRuntimeConfig().public.apiBaseUrl })
</script>
```

nuxt.config
```ts
{
  runtimeConfig: {
    public: { apiBaseUrl: "" }
  }
}
```

.env
```
export NUXT_PUBLIC_API_BASE_URL=http://localhost:12345/api
```

I'd like to know alternative approaches as well.

Support for `.env.dev` / `.env.staging` would be nice.

EDIT: Related https://github.com/nuxt/framework/issues/7646#issue-1377846589

**Comment by gezichenshan** [comment]
@sevillaarvin Thanks for you reply! 

I think we both have the same aim to remove the `{ baseURL: useRuntimeConfig().public.apiBaseUrl }` in every api call.
